File: leetcode/addTwoNumbers.js
class ListNode {
    constructor(val) {
        this.val = val;
        this.next = null;
    }
}

const addTwoNumbers = (list1, list2) => {
    let sum = (list1.val || 0) + (list2.val || 0);
    let currVal = sum % 10;
    let head = new ListNode(currVal);
    let carry = Math.floor(sum / 10);
    let curr = head;

    while (list1.next || list2.next) {
        list1 = list1.next || new ListNode(0);
        list2 = list2.next || new ListNode(0);
        sum = list1.val + list2.val + carry;
        carry = Math.floor(sum / 10);
        currVal = sum % 10;
        curr.next = new ListNode(currVal);
        curr = curr.next;
    }
    if (carry) curr.next = new ListNode(carry);
    return head;
}

const head1 = new ListNode(1);
head1.next = new ListNode(8);
const head2 = new ListNode(0);

console.log(addTwoNumbers(head1, head2));

File: leetcode/addTwoNumbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        carry = 0
        l3 = ListNode()
        curr = l3
        while l1 or l2:
            x = l1.val if l1 is not None else 0
            y = l2.val if l2 is not None else 0
            sum = x + y + carry
            carry = sum // 10
            print(sum)
            curr.next = ListNode(sum % 10)
            l1 = l1.next if l1 is not None else None
            l2 = l2.next if l2 is not None else None
            curr = curr.next
        if carry > 0:
            curr.next = ListNode(carry)
        return l3.next
